bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAdmin.java

```diff
@@ -360,15 +361,78 @@ public void deleteColumnFamily(TableName tableName, byte[] columnName) throws IO
   }
 
   @Override
-  public void modifyTable(TableDescriptor arg0) throws IOException {
-    throw new UnsupportedOperationException("modifyTable"); // TODO
+  public void modifyTable(TableDescriptor tableDescriptor) throws IOException {
+    modifyTable(tableDescriptor.getTableName(), tableDescriptor);
   }
 
   @Override
-  public void modifyTable(TableName arg0, TableDescriptor arg1) throws IOException {
-    throw new UnsupportedOperationException("modifyTable"); // TODO
+  public void modifyTable(TableName tableName, TableDescriptor tableDescriptor) throws IOException {
+    if(isTableAvailable(tableName)) {
+      List<Modification> modifications = new ArrayList<>();
+      buildModificationsForModifyFamilies(tableName,tableDescriptor,modifications);
+      buildModifcationsForDeleteFamilies(tableName,tableDescriptor,modifications);
+      modifyColumn(tableName,"modifyTable","update",(Modification[]) modifications.toArray());
+    }else {
+      throw new TableNotFoundException(tableName);
+    }
   }
 
+  /**
+   * This method will build list of either create or modify Modifications objects..
+   * 
+   * @param tableName
+   * @param tableDescriptor
+   * @param modifications an array of {@link com.google.bigtable.admin.v2.ModifyColumnFamiliesRequest.Modification} object.
+   * @throws IOException
+   */
+  private void buildModificationsForModifyFamilies(TableName tableName, TableDescriptor tableDescriptor,List<Modification> modifications) throws IOException {
```

Comment:
please add a space after ','

Reply:
Also, this code should be in the `bigtable-hbase` project, and ideally a stand alone class.  This logic needs to be used for the HBase 1 `BigtableAdmin` and also the HBase 2 `AsyncAdmin`.